### PR TITLE
Vue cors

### PR DESCRIPTION
**Pull request by Julien-laville**
cosmo
- Switch front to a vue rendered UI
- add Some sugar to UI
- now build with webpack
- add cors (wildcard) header to api

### CODE REVIEW

File: explorer/src/url_state.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+/* init default parameters */
+State = {
+  center : [0,0],
+  zoom : 6,
+  hierarchyId : null
+}
+const urlHash = window.location.hash
+
+if(urlHash && urlHash.split('/').length > 2) { /* parse uri */
```
Comment (Tristramg) [inline explorer/src/url_state.js]: Est-ce que vous avez regardé du côté de vue-router ?

Comment (Julien-laville) [reply explorer/src/url_state.js]: Ni Vue-routeur ni Vuex 
j'ai utilisé le strict minimum de librairies externes

File: api/tilestache.cfg [hunk 2]
```diff
@@ -0,0 +1,27 @@
+{
```
Comment (amatissart) [inline api/tilestache.cfg]: Tilestache is no longer used. This file can be deleted.

File: explorer/src/map/built-style.json [hunk 3]
```diff
@@ -0,0 +1 @@
...ax-width":6.25,"visibility":"visible"},"paint":{"text-halo-blur":1,"text-color":"#334","text-halo-width":2,"text-halo-color":"rgba(255,255,255,0.8)"}},{"id":"place-country-1","type":"symbol","metadata":{"mapbox:group":"1444849242106.713"},"source":"basemap","source-layer":"place","filter":["all",["==","class","country"],["==","rank",1],["has","iso_a2"]],"layout":{"text-font":["Open Sans Bold","Klokantech Noto Sans Bold"],"text-field":"{name}","text-size":{"stops":[[1,11],[4,17]]},"text-transform":"uppercase","text-max-width":6.25,"visibility":"visible"},"paint":{"text-halo-blur":1,"text-color":"#334","text-halo-width":2,"text-halo-color":"rgba(255,255,255,0.8)"}},{"id":"place-continent","type":"symbol","metadata":{"mapbox:group":"1444849242106.713"},"source":"basemap","source-layer":"place","maxzoom":1,"filter":["==","class","continent"],"layout":{"text-font":["Open Sans Bold","Klokantech Noto Sans Bold"],"text-field":"{name}","text-size":14,"text-max-width":6.25,"text-transform":"uppercase","visibility":"visible"},"paint":{"text-halo-blur":1,"text-color":"#334","text-halo-width":2,"text-halo-color":"rgba(255,255,255,0.8)"}}],"id":"cjbdftwmm936t2rquyt8ycvor","visibility":"private"}
```
Comment (antoine-de) [inline explorer/src/map/built-style.json]: I don't think we need this

File: explorer/src/map/sprite.json [hunk 4]
```diff
@@ -0,0 +1 @@
...th":15,"height":15,"x":60,"y":373,"pixelRatio":1},"marker-stroked-11":{"width":15,"height":15,"x":75,"y":373,"pixelRatio":1},"pitch-11":{"width":15,"height":15,"x":90,"y":373,"pixelRatio":1},"ranger-station-11":{"width":15,"height":15,"x":105,"y":373,"pixelRatio":1},"recycling-11":{"width":15,"height":15,"x":120,"y":373,"pixelRatio":1},"residential-community-11":{"width":15,"height":15,"x":135,"y":373,"pixelRatio":1},"roadblock-11":{"width":15,"height":15,"x":150,"y":373,"pixelRatio":1},"shelter-11":{"width":15,"height":15,"x":165,"y":373,"pixelRatio":1},"slaughterhouse-11":{"width":15,"height":15,"x":180,"y":373,"pixelRatio":1},"square-11":{"width":15,"height":15,"x":195,"y":373,"pixelRatio":1},"square-stroked-11":{"width":15,"height":15,"x":210,"y":373,"pixelRatio":1},"star-stroked-11":{"width":15,"height":15,"x":225,"y":373,"pixelRatio":1},"town-11":{"width":15,"height":15,"x":240,"y":373,"pixelRatio":1},"village-11":{"width":15,"height":15,"x":255,"y":373,"pixelRatio":1},"warehouse-11":{"width":15,"height":15,"x":270,"y":373,"pixelRatio":1},"water-11":{"width":15,"height":15,"x":285,"y":373,"pixelRatio":1},"wheelchair-11":{"width":15,"height":15,"x":300,"y":373,"pixelRatio":1}}
```
Comment (antoine-de) [inline explorer/src/map/sprite.json]: same here

File: explorer/src/url_state.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    })
+  }
+}
+
+if(window.location.hash.split('/').length > 3) {
+  State.hierarchyId = parseInt(window.location.hash.split('/')[3])
+}
+
+function update(o) {
+  Object.keys(o).forEach((k) => {
```
Comment (antoine-de) [inline explorer/src/url_state.js]: isn't this just `Object.assign(State, o)` ?

File: explorer/src/vues/hierarchy.vue [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+  import Child from 'vue-loader!./child.vue'
+  import SubHierarchy from 'vue-loader!./sub_hierarchy.vue'
+  export default {
+    name: "hierarchy",
+    components : {
+      Child,
+      SubHierarchy
+    },
+    data : () => {
+      return {
```
Comment (antoine-de) [inline explorer/src/vues/hierarchy.vue]: you can skip this return

File: explorer/src/vues/child.vue [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+
+<script>
+  import Hierarchy from '../hierarchy'
+  export default {
+    name: "Child",
+    props: [
+      'hierarchy'
+    ],
+    data : function () {
+      return {
```
Comment (antoine-de) [inline explorer/src/vues/child.vue]: you can skip this return

**Pull request merged**